File: sorting/insertion_sort/insertion_sort.dart
List<int> insertionSort(List<int> elements) {
  for (var i = 1; i < elements.length; i++) {
    int current = elements[i];
    int j = i - 1;

    while (j >= 0 && current < elements[j]) {
      elements[j + 1] = elements[j];
      j--;
    }

    elements[j + 1] = current;
  }
  return elements;
}

void main() {
  List<int> input = [3, 19, 8, 2, 5, 1];
  print(input);
  print("insertionSort ${insertionSort(List<int>.from(input))}");
}

File: sorting/insertion_sort/insertion_sort.py
def insertion_sort(elements: list[int]) -> list[int]:
    for i in range(1, len(elements)):
        j: int = i-1
        current: int = elements[i]

        while j >= 0 and current < elements[j]:
            elements[j+1] = elements[j]
            j -= 1
        elements[j + 1] = current

    return elements


if __name__ == '__main__':
    input: list[int] = [8, 2, 2, 9, 1]
    print(input)
    print(f"{insertion_sort(input[:]) = }")
